### ISSUE TITLE
[a11y Support] MHV: After Visit Summary and Appointments

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#58536]
## CAIA a11y Support

- [X] Consolidate feedback from Design Intent and related GitHub issue for awareness to AVS - in this ticket. _Completed 5/16/2023_
- [ ] **Receive feedback from AVS** re: current status with feedback.
- [ ] **Schedule meeting with:** Adrian Rollett, Kristen McConnell and Nina Anusavice.
- [ ] [**Wireframe review**](https://www.sketch.com/s/05d5a740-4fc7-4380-8d83-e7acfbab201b) ([requested 5/16/2023](https://dsva.slack.com/archives/C058R5K9XNC/p1684253583506969))

### Existing Related Issue in GitHub
- [Slack Thread](https://dsva.slack.com/archives/C8E985R32/p1678395616652319)
- [Original Ticket Where Issue Surfaced #56282](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56282)
- [OPEN Ticket: Improve appointment list semantic structure #57827](https://github.com/department-of-veterans-affairs/va.gov-team/issues/57827)

> Peter Russo received some advice on how to potentially handle the VoiceOver issue (noted below) so that may not be related. In either case, TalkBack is skipping the links in the scenarios I noted, whether or not they have an aria-label
> Here's the recommendations for how to handle VO reading the link multiple times:
> You have an aria-describedby that’s inside of a web component and shadow dom pointing to an aria-label on a div outside of the web component and shadow dom. That might be okay, but it would be something I’d want to investigate some more.
> you have both an aria label and an aria described by apply did the same thing and they're both announcing.
> I would just get rid of the described by since it is redundant.

[Issue Example 1](https://dsva.slack.com/files/U01C2UUHXH9/F04T244UNSJ/screen_recording_2023-03-09_at_1.09.45_pm.mov)
[Issue Example 2](https://dsva.slack.com/files/U01C2UUHXH9/F04TM780K7B/az_recorder_20230309_075318_edited.mp4)

### Medical Records: Design & Research Collaboration Meeting 5/3/2023
- AVS in appointments first and then do a redirect, since it will be a full year? 
  - **Discussed**: Appointments > Appointments Detail > After Visit Summary > Medical Records at first. 
      - Link to the appointment in the after visit summary itself (name of navigation) > once MR launches, it was discussed that an option could be to pull in the regular breadcrumb. 

- **_(Refer to IA for more on this. Putting here for reference only.)_**

 
### Status from Feedback
- [x] Navigating back solution (ref. Martha)
- [ ] Avoiding disorientation solution (ref. Sara)

### 4/27/2023
  - **Martha W.**: Survey and email should be edited to take into account blind veterans. Once you are in a detail thing, do you need to navigate back? Prefer to open new tabs due AT research. Also,  508 office audit feedback. You told them a new tab, but it was in a new window. 
     _- Future discussion, customizable by facility, per KM 5/19_
     _- It will be in the same window and be able to navigate back per KM 5/19_ 
  - **Sara Sterkenburg:** Rerouted to MHV - the master location even if clicking from detail in appts. The breadcrumbs would reflect the true home of this page. Polyheirarchy if it lives in more than one place, you have to pick a master. Worth talking about with Micki.  These things need to have a master home.  Opening in a new tab, you don’t always know you are and that can create disorientation. 
      _- CAIA KO talked about breadcrumbs with more context and where they came from KM having another conversation with them, per discussion 5/19._ 

### 4/28/2023
**Collab. Cycle: Brian's Ticket: Design Intent - Accessibility Feedback - OCTO Health Apps, MHV Medical Records, After Visit Summary #57660**

### Status of Initial Findings
- [x] Heading hierarchy makes sense
- [x] Alternative solution for the "on this page" component, making the content easier to explore.
- [x] Create a clear understanding of what is contained in each H2 section to help build intuition about the structure of the content - _accordion solution 5/17/2023 @kristenmcconnell implementing, based on office hours with @briandeconinck 

> #### Must
> To the extent that you have control over the after visit content, make sure the heading hierarchy makes sense.
> 
> #### Consider
> Echoing Matt's thoughts on this being a giant massive page of text... I think this would benefit from some kind of design work that makes the content easier to explore. I'm not sure that I love Matt's suggestion of using accordions, but I agree that the existing On This Page component might not be up to the task. Maybe some kind of On This Page type table of contents that's more than just H2s? Some kind of description helper text with each On This Page link to summarize what's in each of those sections?
> 
> My thinking here is that a user is probably not going to have a clear idea of what's in each of those top-level H2 sections without having to do a lot of self-guided exploration. For example, "New orders from this visit" sits under "My treatment plan" which makes sense, but I could imagine expecting to see it under "Today's visit" as well. Anything that can help guide users to the right part of the page faster (or otherwise help build some intuition about the structure of the content) would be valuable.
> 
> I really don't have a specific design recommendation and so long as the heading structure makes sense I don't think there will be any technical WCAG violations. But this might be a good experimental design opportunity for exploring alternatives to or extensions of the On This Page component a bit.

## Action Being Taken
Accordion designs to be semantic when built [(5/17/2023 comment)](https://github.com/department-of-veterans-affairs/va.gov-team/issues/58536#issuecomment-1551719647)

## Related Tickets

- **EPIC:** CAIA support for After Visit Summary and Appointments teams #57423
- **CAIA Intake Ticket:** CAIA Help from After Visit Summary and Appointment Teams #57259

**Comment by sara-amanda** [comment]
@coforma-terry @RLHecht @SarahKay8 @joshkimux @briandeconinck tagging for awareness.

**Comment by RLHecht** [comment]
I believe we can close this ticket but please reopen if anything is still needed.

**Comment by kristenmcconnell** [comment]
Sure. We are finishing our usability sessions tomorrow. I don't think that we have any large findings that will drastically change our direction. I'll reach out if there is anything that we need.
